refatorando-usuarios

Brun0Nasc opened the pull request:
refatorando pro padrão sugerido por matheus

Brun0Nasc committed:
refatorando-usuarios

Brun0Nasc committed:
error-invalid-credentials

matheustgf10 commented on infra/usuarios/postgres/data.go:
```diff
@@ -28,45 +30,128 @@ func (postgres *DBUsuarios) NovoUsuario(req *modelData.Usuario) error {
 
-func (postgres *DBUsuarios) ListarUsuarios(order int) ([]modelApresentacao.ReqUsuario, error) {
-	var sqlStatement string
+func (pg *DBUsuarios) ListarUsuarios(order int) (res *modelApresentacao.ListaUsuarios, err error) {
```
Assim ficaria com a sugestão que dei mais cedo, no caso precisaria de duas novas funções nao pacote `utils` e precisaria que na requisição fosse passada os dois parametros pelo query. Ex.: `orderby=nome` e `order=desc`

![codeRequestParamsData](https://user-images.githubusercontent.com/30707765/192068538-2464ae2f-c083-478a-a9b9-a3a0b136c7fb.png)

matheustgf10 replied:
Com isso teria que ser feita uma refatoração desde o handler.go até o data, pois o parametro deixaria de ser um inteiro e passaria a ser uma estrutura `RequestParams` [também criado no novo arquivo do pacote **utils**].

matheustgf10 replied:
Assim ficaria o novo arquivo da **utils**. Uma função para percorrer todos os parametros informados na requisição, e uma função do próprio **params**
![codeRequestParamsUtils](https://user-images.githubusercontent.com/30707765/192068771-45c128e2-0392-4972-8c9d-f62d9bc515d3.png)

Brun0Nasc committed:
Implementando funções para testar os parâmetros (incompleto)

A fazer - Testar listagem de usuários inativados; Alinhar as mudanças ao handler;

Brun0Nasc committed:
refatorando usuários (incompleto)

matheustgf10 commented on infra/usuarios/interface.go:
```diff
@@ -5,8 +5,8 @@ import(
-	NovoUsuario(req *modelApresentacao.ReqUsuario) error
-	ListarUsuarios(order int) ([]modelApresentacao.ReqUsuario, error)
+	NovoUsuario(req *modelApresentacao.Usuario) error
+	ListarUsuarios(order int) (*modelApresentacao.ListaUsuarios, error)
```
```suggestion
	ListarUsuarios(params *utils.RequestParams) (*modelApresentacao.ListaUsuarios, error)
```

matheustgf10 commented on domain/usuarios/usecase.go:
```diff
@@ -23,7 +23,7 @@ func NovoUsuario(req *modelApresentacao.ReqUsuario) (err error) {
 }
 
-func ListarUsuarios(order int) (res []modelApresentacao.ReqUsuario, err error) {
+func ListarUsuarios(order int) (res *modelApresentacao.ListaUsuarios, err error) {
```
```suggestion
func ListarUsuarios(params *utils.RequestParams) (res *modelApresentacao.ListaUsuarios, err error) {
```

matheustgf10 replied:
e também no handler.go

matheustgf10 reviewed (changes requested):
Tem que fazer as mudanças aqui também se não a build quebra

matheustgf10 committed:
[FQC-3] refactor: modificando a forma de listar usuários inativos e usuários ativos

matheustgf10 commented:
Os problemas relatados acima foram corrigidos na #4.

matheustgf10 reviewed (approved):
A **[FQC-3]** Deve ser mergeada na **[FQC-2]** antes que essa seja mergeada na **main**

Brun0Nasc committed:
Mudando Frequencia (Agora frequencia-usuario)

Brun0Nasc committed:
mudando-frequencia

Brun0Nasc committed:
Mudando Frequencia

Brun0Nasc committed:
update main.go

Brun0Nasc committed:
Merge branch 'FQC-4' of https://github.com/Brun0Nasc/Frequencias-Backend into FQC-4

Brun0Nasc committed:
Update dbpostgres.go

Brun0Nasc committed:
refactor: model usuarios - novo campo: removed_at

Brun0Nasc committed:
Merge pull request #5 from Brun0Nasc/FQC-4

FQC-3+FQC-4

Brun0Nasc committed:
Merge pull request #4 from Brun0Nasc/FQC-3

[FQC-3] Modificando forma de listar usuários ativos e inativos através de filtros

Brun0Nasc committed:
Fazendo o filtro

matheustgf10 commented:
:eyes: tá ficando muito bom, parabéns guys!! :clap: :rocket:

Brun0Nasc committed:
listagem de frequencias por data

listagem de frequencias de uma data específica agora funcionando

Brun0Nasc committed:
Merge pull request #7 from Brun0Nasc/FQC-5

Fqc 5

Brun0Nasc committed:
Merge branch 'main' into FQC-2
